### PR TITLE
Bug 1159411 - Allow ability to set a root folder

### PR DESCRIPTION
**Pull request by daleharvey**
Adds the argparse module, the argparsing is fairly messy and should be cleaned up but this is backwards compatible

### CODE REVIEW

File: preload.py [hunk 1]
```diff
@@ -321,12 +322,21 @@ def fetch_webapp(app_url, directory=None):
 
 
 def main():
+
+    parser = argparse.ArgumentParser(description='Preload webapps.')
```
Comment (gasolin) [inline preload.py]: keep space between ` = ` & strings

File: preload.py [hunk 2]
```diff
@@ -321,12 +322,21 @@ def fetch_webapp(app_url, directory=None):
...
+    parser.add_argument('--root', default=os.getcwd(),
+                        help='Root directory to run from')
```
Comment (gasolin) [inline preload.py]: same here

File: preload.py [hunk 3]
```diff
@@ -321,12 +322,21 @@ def fetch_webapp(app_url, directory=None):
...
+    args = parser.parse_args()
+
+    os.chdir(args.root)
+    print args.root
+
     # icon convertion script
     if (len(sys.argv) == 3 and sys.argv[1] == "--icon"):
```
Comment (gasolin) [inline preload.py]: if you prefer `argparse`, please migrate all options to `argparse`.

File: preload.py (hunk 2 above)
Comment (gasolin) [inline preload.py]: and please add a line of document in `README.md` to demonstrate how this option works :notebook:

**Comment by gasolin** [comment]
test still fail

ERROR: test_fetch_resource (__main__.PreloadTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/gasolin/Documents/web/sparkle/gaia-preload-app/.env/lib/python2.7/site-packages/mock.py", line 1201, in patched
    return func(*args, **keywargs)
  File "preload_test.py", line 102, in test_fetch_resource
    'local_dir/', 'test.manifest')
  File "/Users/gasolin/Documents/web/sparkle/gaia-preload-app/preload.py", line 142, in fetch_resource
    except urllib.URLError as e:
AttributeError: 'module' object has no attribute 'URLError'

======================================================================
FAIL: test_fetch_appcache (__main__.PreloadTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/gasolin/Documents/web/sparkle/gaia-preload-app/.env/lib/python2.7/site-packages/mock.py", line 1201, in patched
    return func(*args, **keywargs)
  File "preload_test.py", line 152, in test_fetch_appcache
    assert newlines[0] == 'CACHE MANIFEST'
AssertionError

**Comment by daleharvey** [comment]
Ok reran the tests with + without virtualenv, all passing here

**Pull request merged**